estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,62 @@
+import logging
+
+from odoo import api, fields, models, tools
+from odoo.exceptions import UserError
+
+logger = logging.getLogger(__name__)
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Property Offer"
+    _order = "price desc"
+    _sql_constraints = [
+        ('check_price', 'CHECK(price > 0)',
+         'An offer price must be strictly positive'),
+    ]
+
+    price = fields.Float(required=True)
+    status = fields.Selection(selection=[
+        ('accepted', 'Accepted'),
+        ('refused', 'Refused'),
+    ], copy=False)
+    
+    partner_id = fields.Many2one("res.partner", required=True)
+    property_id = fields.Many2one("estate.property", required=True)
+    validity = fields.Integer("Validity (days)", default=7)
+    date_deadline = fields.Date("Deadline", compute="_compute_date_deadline", inverse="_inverse_date_deadline")
+    property_type_id = fields.Many2one(related="property_id.property_type_id", store=True)
+    
+    @api.depends("validity")
+    def _compute_date_deadline(self) -> None:
+        for offer in self:
+            offer.date_deadline = fields.Datetime.add(offer.create_date or fields.Datetime.now(), days=offer.validity)
+            
+    def _inverse_date_deadline(self) -> None:
+        for offer in self:
+            offer.validity = (offer.date_deadline -  offer.create_date.date()).days
+            
+    @api.model
+    def create(self, vals):
+        property = self.env["estate.property"].browse(vals["property_id"])
+        if tools.float_compare(vals["price"], property.best_price, 0) == -1:
+            raise UserError("Offer with lower amount than an existing offer!")
+        property.state = "offer_received"
+        return super().create(vals)
+            
+    def action_offer_accepted(self) -> bool:
+        if any(o.status == "accepted" for o in self.property_id.offer_ids):
```

Comment:
```suggestion
        if any(offer.status == "accepted" for offer in self.property_id.offer_ids):
```
I feel like offer instead of o is not much more effort, but much more clear. It doesn't matter much for one line, but it tends to spread.